packages/frontend/src/features/semanticViewer/components/SemanticViewerVizConfig/CartesianVizFieldConfig.tsx

```diff
@@ -0,0 +1,300 @@
+        field !== undefined &&
+        !groupByOptions.find((x) => x.reference === field.reference)
+            ? `Column "${field.reference}" does not exist. Choose another`
+            : undefined;
+    return (
+        <FieldReferenceSelect
+            rightSection={
+                // When the field is deleted, the error state prevents the clear button from showing
+                error && (
+                    <ActionIcon
+                        onClick={() =>
+                            dispatch(actions.unsetGroupByReference())
+                        }
+                    >
+                        <MantineIcon icon={IconX} />
+                    </ActionIcon>
+                )
+            }
+            clearable
+            data={groupByOptions.map((groupBy) => ({
+                value: groupBy.reference,
+                label: groupBy.reference,
+            }))}
+            value={field?.reference ?? null}
+            placeholder="Select group by"
+            error={error}
+            onChange={(value) => {
+                if (!value) {
+                    dispatch(actions.unsetGroupByReference());
+                } else {
+                    dispatch(
+                        actions.setGroupByReference({
+                            reference: value,
+                        }),
+                    );
+                }
+            }}
+            fieldType={
+                columns?.find((x) => x.reference === field?.reference)?.type ??
+                DimensionType.STRING
+            }
+        />
+    );
+};
+
+export const CartesianVizFieldConfig = ({
+    columns,
+    actions,
+    selectedChartType,
+}: {
+    selectedChartType: ChartKind;
+    columns: SemanticLayerColumn[];
+
+    actions: CartesianChartActionsType;
+}) => {
+    const dispatch = useVizDispatch();
+    const indexOptions = useVizSelector((state) =>
+        cartesianChartSelectors.getIndexLayoutOptions(state, selectedChartType),
+    );
+    const valueOptions = useVizSelector((state) =>
+        cartesianChartSelectors.getValuesLayoutOptions(
+            state,
+            selectedChartType,
+        ),
+    );
+    const xAxisField = useVizSelector((state) =>
+        cartesianChartSelectors.getXAxisField(state, selectedChartType),
+    );
+    const yAxisFields = useVizSelector((state) =>
+        cartesianChartSelectors.getYAxisFields(state, selectedChartType),
+    );
+    const groupByField = useVizSelector((state) =>
+        cartesianChartSelectors.getGroupByField(state, selectedChartType),
+    );
+    const groupByLayoutOptions = useVizSelector((state) =>
+        cartesianChartSelectors.getPivotLayoutOptions(state, selectedChartType),
+    );
+
+    // For axis layout options, we get the available options from the
+    // viz slice, then we have to filter them with the extra info (kind)
+    // from the columns.
+    const xLayoutOptions = useMemo(
+        () =>
+            indexOptions?.filter((option) => {
```

Comment:
This filtering is a little bit awkward. We get generic fields (no 'kind') from the chart slice and have to filter them by the fields. I'm open to better ideas :D